code/game/mecha/equipment/tools/mining_tools.dm

```diff
@@ -118,13 +130,10 @@
 		var/obj/mecha/working/mecha = loc
 		if(!mecha.occupant)
 			return
-		var/mob/pilot = mecha.occupant
-		var/list/occupant = list()
-		occupant |= pilot
-		pilot.sight |= SEE_TURFS
+		var/list/L = list(mecha.occupant)
 		scanning = 1
-		mineral_scan_pulse(occupant,get_turf(loc))
+		mineral_scan_pulse(L,get_turf(loc))
```

Comment:
The first argument of mineral_scan_pulse() must be a list. Not sure if this was causing bugs.